### PR TITLE
Replace uint by usize in libcore/panicking.rs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #22240.

### CODE REVIEW

**Comment by huonw** [comment]
I suspect most of these should be `u32`: the memory size of the target architecture doesn't relate to the size of the source on the host.

File: src/libcore/panicking.rs [hunk 1]
```diff
@@ -34,25 +34,25 @@ use fmt;
 
 #[cold] #[inline(never)] // this is the slow path, always
 #[lang="panic"]
-pub fn panic(expr_file_line: &(&'static str, &'static str, uint)) -> ! {
+pub fn panic(expr_file_line: &(&'static str, &'static str, usize)) -> ! {
```
Comment (aturon) [inline src/libcore/panicking.rs]: This should be either `u32` or `u64`, you could make an argument for either -- I'd suggest `u64` based on the maximum file length we report and the lack of performance issues here.

**Comment by GuillaumeGomez** [comment]
I was speaking with @aturon about the argument size, I was more for u32 but u64 seems better.

**Comment by pnkfelix** [comment]
cc #22240

**Comment by GuillaumeGomez** [comment]
I updated the code. Any thoughts ? @aturon, @huonw ?

**Comment by alexcrichton** [comment]
I believe this actually ended up getting taken care of in https://github.com/rust-lang/rust/pull/22853 actually, but thanks for the PR!

**Pull request closed without merging**